lib/gradle/parse/buildGradleParser.ts

```diff
@@ -69,7 +75,77 @@ export async function getProjectGroup(p: Project): Promise<string> {
         }
     }
     return group;
+}
+
+export async function getGradleModules(p: Project): Promise<string[]> {
+    if (p.hasFile("settings.gradle")) {
+        const pattern = /include ["']([a-zA-Z0-9\-:]+)["']/;
+        const settingsContent = await (await p.getFile("settings.gradle")).getContent();
+        const matches = settingsContent.split(/\r?\n/)
+            .map(line => line.match(pattern))
+            .filter(m => m && m.length > 0)
+            .map(m => m[1]);
+        return matches;
+    }
+    return Promise.reject("Not a multimodule project");
+}
+
+export async function getRuntimeClasspath(p: LocalProject, module?: string): Promise<string[]> {
+    const initScript = `
+allprojects {
+    task generateRuntimeClasspath {
+        def output = new File(rootDir, ".atomist-dependencies.txt");
+
+        doLast {
+            output.text = configurations.runtimeClasspath.resolve()
+        }
+    }
+}
 
+    `;
+    const log = new StringCapturingProgressLog();
+    p.addFileSync(".atomist.gradle", initScript);
+    if (module) {
+        await spawnLog(await determineGradleCommand(p),
+            ["--init-script", ".atomist.gradle", "-q", `:${module}:generateRuntimeClasspath`], {log, cwd: p.baseDir});
+    } else {
+        await spawnLog(await determineGradleCommand(p),
+            ["--init-script", ".atomist.gradle", "-q", `:generateRuntimeClasspath`], {log, cwd: p.baseDir});
+    }
+    p.deleteFileSync(".atomist.gradle");
+    const output = await (await p.getFile(".atomist-dependencies.txt")).getContent();
+    p.deleteFileSync(".atomist-dependencies.txt");
+    const dependencies = output.substring(1, output.length - 1);
+    return dependencies.split(", ");
+}
+
+export async function getCompileClasspath(p: LocalProject, module?: string): Promise<string[]> {
```

Comment:
Long term it's probably best to expose the individual configurations. `getClasspathForConfiguration(p, "somemodule", "testCompile")`. 

Perhaps it may even be worth adding a `getConfigurations`. Useful utility methods I can imagine there are things like `hasDependency`,

Reply:
That sounds useful, indeed.